### ISSUE TITLE
Why can't I choose a custom form class for formset forms?

### ISSUE THREAD

**Issue by helderco** [wagtail/django-modelcluster#125]
Hi, I'm trying to use django-modelcluster in the frontend as well. 

I have some views that have formsets created with inlineformset_factory(), but I like the way ClusterForm has the ability to handle child formsets so I'm trying to clean up the view code by extending ClusterForm.

The issue I'm running is that normally, when I create a formset, I can set the base class for the formset's forms, but not with ClusterForm. The reason is in the metaclass:

```python
class ClusterFormMetaclass(ModelFormMetaclass):
    @classmethod
    def child_form(cls):
        return ClusterForm

    def __new__(cls, name, bases, attrs):
        ...
            ...
                ...
                kwargs = {
                    'extra': cls.extra_form_count,
                    'formfield_callback': formfield_callback,
                    'fk_name': rel.field.name,
                    'widgets': widgets
                }

                # see if opts.formsets looks like a dict; if so, allow the value
                # to override kwargs
                try:
                    kwargs.update(opts.formsets.get(rel_name))
                except AttributeError:
                    pass

                formset = childformset_factory(opts.model, rel.field.model, form=cls.child_form(), **kwargs)
                ....
```

Why not allow the kwargs.update() to override form by moving `form=cls.child_form()` to `kwargs = {...., 'form': cls.child_form(), ...}`?

Or is there a better way?

Thanks!

**Comment by gasman** [comment]
@helderco Thanks for the suggestion! This seems good to me - I'd be happy to accept a PR that makes this change.

**Comment by gasman** [comment]
Implemented in #126
